### DISCUSSION TITLE
New ModelCheckpoint Code Update Breaking Model Saving Functionality : v1.2.5->v.1.2.6

### DISCUSSION THREAD

**Discussion by valayDave** [Lightning-AI/pytorch-lightning#6769] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am using Latest PT-Lightning and your upgrade to [ModelCheckpoint](https://github.com/PyTorchLightning/pytorch-lightning/blob/f0c5479de99e9a1524c95a9dffe2fa0cc76ab1ea/pytorch_lightning/callbacks/model_checkpoint.py) broke my functionality to my codebase. 

V1.2.6 specifically coz I started facing these issues yesterday.  Downgraded to V1.2.5. Please check once. The dict based metrics returned from the `train_step` or `validation_step` are not captured in the ModelCheckpointCallback 

```python 
 monitor_candidates = self._monitor_candidates(trainer)
```
`monitor_candidates` doesn't have any values returned from the trainer. And I am not using the `Result` class. I am returning a pure dictionary. 

Didn't have time to create replicate the bug exactly in a notebook because I am on a tight deadline. But please do check. 

As I have more time, I will update more on the issue.

**Comment by awaelchli** [comment, marked as answer]
You need to use `self.log("your_metric", metric)`
And then set the monitor argument in `ModelCheckpoint(monitor="your_metric")`

Let me know if that helps.

**Reply by carmocca** [reply]
Exactly. Setting monitor candidates through dictionary returns is legacy behaviour and no longer supported.
